### PR TITLE
chore:  WIP documentation improvements

### PR DESCRIPTION
**Pull request by GordonSmith**
Signed-off-by: Gordon Smith <GordonJSmith@gmail.com>

## Checklist:
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit message includes a "fixes" reference if appropriate.
  - [x] The commit is signed.
- [ ] The change has been fully tested:
  - [ ] I have viewed all related gallery items
  - [ ] I have viewed all related dermatology items
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised new issues to address them separately

### CODE REVIEW

File: components/core/src/react.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
-import type { CSSProperties, DOMAttributes } from "react";
+import type { ClassAttributes, CSSProperties, DOMAttributes, Props } from "react";
```
Comment (jbrundage) [inline components/core/src/react.ts:1]: Props is defined but not used?

File: components/layout/src/zoom.ts [hunk 2]
```diff
@@ -60,7 +60,8 @@ export class HPCCZoomElement extends HPCCSVGElement {
             this.x = transform.x;
             this.y = transform.y;
             this.scale = transform.k;
-            this.$emit("changed");
+            this.$emit("change");
+            this.$emit("Change");
```
Comment (jbrundage) [inline components/layout/src/zoom.ts:64]: Is this debounced?

File: components/preview/README.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+```html
+<head>
+    <script src="https://cdn.jsdelivr.net/npm/@hpcc-js/wc-preview"></script>
+</head>
+
+<body>
+    <hpcc-preview>
+        ...
+    </hpcc-preview>
+</body>
```
Comment (jbrundage) [inline components/preview/README.md:50]: Maybe these issues will be resolved when these changes are published?

The ES Module preview throws this error: Uncaught Error: [Package Error] "tslib" no dependency version info found. (Imported by "@hpcc-js/wc-core").

The UMD preview throws a 404 error for this file: "assets/index.umd.min.js"

File: components/preview/package.json [hunk 4]
```diff
@@ -20,15 +15,19 @@
...
+    "compile-es6-watch": "npm run compile-es6 -- -w",
     "bundle": "rollup -c ../rollup.config.js",
+    "bundle-watch": "npm run bundle -- -w",
     "build": "npm run compile-es6 && npm run bundle",
     "update": "npx npm-check-updates -u -t minor"
   },
   "dependencies": {
     "@hpcc-js/wc-core": "^0.2.0",
     "@hpcc-js/wc-editor": "^0.2.0",
+    "@hpcc-js/wc-layout": "^0.1.0",
```
Comment (jbrundage) [inline components/preview/package.json:27]: This doesn't match the `"@hpcc-js/wc-layout"` version in `layout/package.json`

File: components/preview/src/preview.ts [hunk 5]
```diff
@@ -33,61 +29,52 @@ export class HPCCPreviewElement extends HPCCResizeElement {
...
         super.enter();
-        this._head = document.head.innerHTML.toString();
-        const codeElements = this.getElementsByTagName("code")[0];
-        this._cm.text = codeElements?.innerText ?? "" + this.innerHTML;
-        this.gatherScripts(document.body, this._scripts);
-        this._cm.addEventListener("change", () => {
-            this.content = this._cm.text.trim();
-        });
+        this.slotChanged();
+        this._slot.addEventListener("slotchange", () => this.slotChanged());
```
Comment (jbrundage) [inline components/preview/src/preview.ts:74]: Is this event ever emitted?

File: components/preview/src/vitepress.ts [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+     * Content to be displayed in the preview iframe
+     * 
+     * @defaultValue ""
+     */
+    @attribute content = "";
+
+    /**
+     * Split ratio between the preview and the code editor
+     * 
+     * @defaultValue 1/3
```
Comment (jbrundage) [inline components/preview/src/vitepress.ts:52]: Shouldn't this be `2 / 3`?

File: demos/react/typescript/package.json [hunk 7]
```diff
@@ -0,0 +1,46 @@
+{
+  "name": "@hpcc-js/demo-react-typescript",
+  "version": "0.1.0",
+  "private": true,
+  "dependencies": {
+    "@hpcc-js/wc-layout": "^0.1.0",
```
Comment (jbrundage) [inline demos/react/typescript/package.json:6]: This doesn't match the layout/package.json version `0.0.0`

File: components/layout/src/zoom.ts (hunk 2 above)
Comment (GordonSmith) [reply components/layout/src/zoom.ts:64]: No - but there should only be 1 emit (I was testing against React and trying to get it to work with synthetic events).

**Review by jbrundage — APPROVED** [review]

File: components/preview/src/vitepress.ts (hunk 6 above)
Comment (GordonSmith) [reply components/preview/src/vitepress.ts:52]: Maybe - am planning on replacing with the layout splitpanel anyway

File: components/preview/README.md (hunk 3 above)
Comment (GordonSmith) [reply components/preview/README.md:50]: The work on hpcc-preview is just starting.

File: components/preview/package.json (hunk 4 above)
Comment (GordonSmith) [reply components/preview/package.json:27]: Will fix.

Comment (GordonSmith) [reply components/preview/package.json:27]: Resolved

File: components/layout/src/zoom.ts (hunk 2 above)
Comment (GordonSmith) [reply components/layout/src/zoom.ts]: Resolved

File: demos/react/typescript/package.json (hunk 7 above)
Comment (GordonSmith) [reply demos/react/typescript/package.json:6]: Resolved

File: components/layout/package.json [hunk 8]
```diff
@@ -20,18 +15,21 @@
...
     "compile-es6": "tsc -p ./tsconfig.json",
+    "compile-es6-watch": "npm run compile-es6 -- -w",
     "bundle": "rollup -c ../rollup.config.js",
+    "bundle-watch": "npm run bundle -- -w",
     "build": "npm run compile-es6 && npm run bundle",
     "update": "npx npm-check-updates -u -t minor"
   },
   "dependencies": {
-    "@hpcc-js/wc-core": "^0.2.0",
+    "@hpcc-js/wc-core": "^0.1.1",
```
Comment (jbrundage) [inline components/layout/package.json:25]: Why `0.1.1` here but nowhere else?

**Review by jbrundage — APPROVED** [review]

File: components/layout/package.json (hunk 8 above)
Comment (GordonSmith) [reply components/layout/package.json]: Ok - I resynced all versions...

File: components/core/src/react.ts (hunk 1 above)
Comment (GordonSmith) [reply components/core/src/react.ts]: Removed

**Review by jbrundage — APPROVED** [review]

**Pull request merged**